src/python/pants/engine/internals/selectors.py

```diff
@@ -233,183 +227,172 @@ def __await__(self) -> Generator[Tuple[Get, ...], None, Tuple]:
 # These type variables are used to parameterize from 1 to 10 Gets when used in a tuple-style
 # MultiGet call.
 
-_P0 = TypeVar("_P0")
-_P1 = TypeVar("_P1")
-_P2 = TypeVar("_P2")
-_P3 = TypeVar("_P3")
-_P4 = TypeVar("_P4")
-_P5 = TypeVar("_P5")
-_P6 = TypeVar("_P6")
-_P7 = TypeVar("_P7")
-_P8 = TypeVar("_P8")
-_P9 = TypeVar("_P9")
-
-_SDT0 = TypeVar("_SDT0")
-_SDT1 = TypeVar("_SDT1")
-_SDT2 = TypeVar("_SDT2")
-_SDT3 = TypeVar("_SDT3")
-_SDT4 = TypeVar("_SDT4")
-_SDT5 = TypeVar("_SDT5")
-_SDT6 = TypeVar("_SDT6")
-_SDT7 = TypeVar("_SDT7")
-_SDT8 = TypeVar("_SDT8")
-_SDT9 = TypeVar("_SDT9")
-
-_S0 = TypeVar("_S0")
-_S1 = TypeVar("_S1")
-_S2 = TypeVar("_S2")
-_S3 = TypeVar("_S3")
-_S4 = TypeVar("_S4")
-_S5 = TypeVar("_S5")
-_S6 = TypeVar("_S6")
-_S7 = TypeVar("_S7")
-_S8 = TypeVar("_S8")
-_S9 = TypeVar("_S9")
+_Out0 = TypeVar("_Out0")
+_Out1 = TypeVar("_Out1")
+_Out2 = TypeVar("_Out2")
+_Out3 = TypeVar("_Out3")
+_Out4 = TypeVar("_Out4")
+_Out5 = TypeVar("_Out5")
+_Out6 = TypeVar("_Out6")
+_Out7 = TypeVar("_Out7")
+_Out8 = TypeVar("_Out8")
+_Out9 = TypeVar("_Out9")
+
+_In0 = TypeVar("_In0")
+_In1 = TypeVar("_In1")
+_In2 = TypeVar("_In2")
+_In3 = TypeVar("_In3")
+_In4 = TypeVar("_In4")
+_In5 = TypeVar("_In5")
+_In6 = TypeVar("_In6")
+_In7 = TypeVar("_In7")
+_In8 = TypeVar("_In8")
+_In9 = TypeVar("_In9")
```

Comment:
This is great too!!!

Reply:
Credit to John for this. It's to support heterogeneous `MultiGet`, which was a huge improvement to rule authoring because it's now ergonomic to batch multiple requests in a single `await`.